src/contracts/GPv2Settlement.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-license-identifier: LGPL-3.0-or-newer
+pragma solidity ^0.6.12;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@uniswap/v2-core/contracts/interfaces/IUniswapV2Factory.sol";
+
+/// @title Gnosis Protocol v2 Settlement Contract
+/// @author Gnosis Developers
+contract GPv2Settlement {
+    /// @dev The domain separator used for signing orders that gets mixed in
+    /// making signatures for different domains incompatible.
+    bytes32 public constant DOMAIN_SEPARATOR = keccak256("GPv2");
+
+    /// @dev The stride of an encoded order.
+    uint256 private constant ORDER_STRIDE = 120;
+
+    /// @dev The Uniswap factory. This is used as the AMM that GPv2 settles with
+    /// and is responsible for determining the range of the settlement price as
+    /// well as trading surplus that cannot be directly settled in a batch.
+    IUniswapV2Factory public immutable uniswapFactory;
+
+    /// @dev The global nonce for determining the current batch. This is used to
+    /// ensure orders can't be replayed in multiple batches.
+    uint256 public nonce;
```

Comment:
This should be a mapping, as we for sure want a global nonce per pair

Reply:
:see_no_evil:

Reply:
So, I did something I **think** is cleaver for the nonce mapping, let me know if you see any issues with it. Maybe @fedgiac as our resident cryptographer takes issue to my wild-wild-west Sftw.Eng. antics.